src/SprykerEco/Zed/AmazonQuicksight/Communication/Controller/AnalyticsController.php

```diff
@@ -0,0 +1,116 @@
+        $enableAnalyticsForm->handleRequest($request);
+
+        if (!$enableAnalyticsForm->isSubmitted()) {
+            return $this->viewResponse([
+                'enableAnalyticsForm' => $enableAnalyticsForm->createView(),
+            ]);
+        }
+
+        if (!$enableAnalyticsForm->isValid()) {
+            $this->addErrorMessage('CSRF token is not valid');
+        }
+
+        $enableQuicksightAnalyticsResponseTransfer = $this->getFacade()->enableAnalytics(
+            (new EnableQuicksightAnalyticsRequestTransfer())
+                ->setAssetBundleImportJobId($this->getFactory()->getConfig()->getDefaultAssetBundleImportJobId())
+                ->setUser($this->findCurrentUser()),
+        );
+
+        foreach ($enableQuicksightAnalyticsResponseTransfer->getErrors() as $errorTransfer) {
+            $this->addErrorMessage($errorTransfer->getMessageOrFail());
+        }
+
+        return $this->redirectResponse(static::URL_ANALYTICS);
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|array<string, mixed>
+     */
+    public function resetAction(Request $request): RedirectResponse|array
+    {
+        $resetAnalyticsForm = $this->getFactory()->getResetAnalyticsForm();
+        $resetAnalyticsForm->handleRequest($request);
+
+        if (!$resetAnalyticsForm->isSubmitted()) {
+            return $this->viewResponse([
+                'resetAnalyticsForm' => $resetAnalyticsForm->createView(),
+            ]);
+        }
+
+        if (!$resetAnalyticsForm->isValid()) {
+            $this->addErrorMessage('CSRF token is not valid');
+        }
+
+        $resetQuicksightAnalyticsResponseTransfer = $this->getFacade()->resetAnalytics(
+            (new ResetQuicksightAnalyticsRequestTransfer())
+                ->setAssetBundleImportJobId($this->getFactory()->getConfig()->getDefaultAssetBundleImportJobId())
+                ->setUser($this->findCurrentUser()),
+        );
+
+        foreach ($resetQuicksightAnalyticsResponseTransfer->getErrors() as $errorTransfer) {
+            $this->addErrorMessage($errorTransfer->getMessageOrFail());
+        }
+
+        return $this->redirectResponse(static::URL_ANALYTICS);
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\UserTransfer|null
+     */
+    protected function findCurrentUser(): ?UserTransfer
+    {
+        $userFacade = $this->getFactory()->getUserFacade();
+        $userTransfer = $userFacade->getCurrentUser();
+        $userCollectionTransfer = $userFacade->getUserCollection(
+            (new UserCriteriaTransfer())->setUserConditions(
+                (new UserConditionsTransfer())->addIdUser($userTransfer->getIdUserOrFail()),
+            ),
+        );
+
+        return $userCollectionTransfer->getUsers()->offsetExists(0)
+            ? $userCollectionTransfer->getUsers()->offsetGet(0) :
+            null;
```

Comment:
`...->getIterator()->current()` will also return `null` if a collection is empty.